### PR TITLE
When using asymmetric encryption a kid header is convenient so JSON Web Key Sets can be used 

### PR DESCRIPTION
**Pull request by vanpelt**
Ideally loginsrv would just create a JSON Web Key Set and serve it up, but this is a a quick and easy fix for now.  This was inspired by [Auth0](https://auth0.com/docs/jwks)

### CODE REVIEW

**Comment by g-w** [comment]
Hello @vanpelt, 

thank you for your contribution! I will merge your PR if you add a tests that checks that the `kid` Header is set. 

My understanding of JWK specifies a data structure that holds a key respectively a JWK Set a datastructure that holds a set of keys. So it is more like PEM/DER. I currently don't understand how the changes are related to JWK Sets, but optionally adding the key id to the headers seems useful to me. Could you please rephrase the description of the PR stating what is actually changed and (if you like) open an issue that we might support JWK Sets?

**Review by g-w — CHANGES_REQUESTED** [review]
Add test for handler.

**Comment by kernle32dll** [comment]
I picked up the ball at this, and have made the necessary adjustments. Just waiting for other things to get merged first.

I compiled this change amongst other things in my own master - you can take it for a spin here: https://hub.docker.com/r/kernle32dll/loginsrv

**Pull request closed without merging**
